File: cpp/leetcode/043.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        int m = num1.size(), n = num2.size();
        string res(m + n, '0');
        for(int i = m - 1; i>=0; i--){
            for(int j = n - 1; j>=0; j--){
                int tmp = res[i+j+1] - '0' + (num1[i] - '0') * (num2[j] - '0');
                res[i+j+1] = tmp % 10 + '0';
                res[i+j] += tmp / 10;
            }
        }
        auto idx = res.find_first_not_of('0');
        if(idx == string::npos) return "0";
        else return res.substr(idx);
    }
};

File: python/leetcode/043.py
class Solution:
    def multiply(self, num1, num2):
        n, m = len(num1), len(num2)
        ans = [0 for i in range(n + m)]
        for i in range(n - 1, -1, -1):
            for j in range(m - 1, -1, -1):
                mul = int(num1[i]) * int(num2[j])
                idx1 = i + j
                idx2 = i + j + 1
                ans[idx1] += (mul + ans[idx2]) / 10
                ans[idx2] = (mul + ans[idx2]) % 10
        res =  ''.join(map(str, ans)).lstrip("0")
        return res if res else "0"